src/util/exportUtils.js

```diff
@@ -2,84 +2,37 @@ const { BadRequestError } = require('./errorUtils');
-    .map(function (typeFilter) {
-      logger.debug(`Adding typeFilter ${typeFilter} to exportUrl typeFilter parameter`);
-      return typeFilter.valueString;
-    })
-    .toString();
-
-  if (typesString) {
-    if (exportUrl.includes(`_type=`)) {
-      console.warn('_type parameter already supplied in exportUrl. Omitting entries from parameter array');
-    } else {
-      // add types from parameters to exportUrl
-      exportUrl += `${exportUrl.includes('_typeFilter=') ? '&' : '?'}_type=${typesString}`;
-    }
-  }
-
-  if (typeFilterString) {
-    if (exportUrl.includes(`_typeFilter=`)) {
-      console.warn('_typeFilter parameter already supplied in exportUrl. Omitting entries from parameter array');
-    } else {
-      // add type filters from parameters to exportUrl
-      exportUrl += `${exportUrl.includes('_type=') ? '&' : '?'}_typeFilter=${typeFilterString}`;
-    }
-  }
-  return exportUrl;
+const retrieveInputUrls = parameters => {
+  logger.debug(`Retrieving all input URLs from parameters: ${JSON.stringify(parameters)}`);
+  const inputParamArray = parameters.filter(param => param.name === 'input');
+  checkInputUrlArray(inputParamArray);
+
+  const inputUrlArray = inputParamArray.flatMap(param =>
+    param.part
+      .filter(p => p.name === 'url')
+      .map(part => ({ type: part.valueUrl.split('.ndjson')[0].split('/').at(-1), url: part.valueUrl }))
+  );
+  return inputUrlArray;
 };
 
 /**
- * Checks whether the export URL array contains exactly one exportUrl
- * @param {Array} exportUrlArray array of export URLs provided in request
+ * Checks whether the input URL array contains at least one input url and that
+ * it contains a valueUrl
+ * @param {Array} inputUrlArray array of input URLs provided in the request
  */
-const checkExportUrlArray = exportUrlArray => {
-  if (exportUrlArray.length === 0) {
-    throw new BadRequestError(`No exportUrl parameter was found.`);
-  }
-  if (exportUrlArray.length !== 1) {
-    throw new BadRequestError(`Expected exactly one export URL. Received: ${exportUrlArray.length}`);
-  }
-  // if one export URL exists, check that valueUrl exists
-  if (!exportUrlArray[0].valueUrl) {
-    throw new BadRequestError(`Expected a valueUrl for the exportUrl, but none was found`);
-  }
-};
-
-/**
- * Uses request body parameter to search for the export server type if there is one.
- * If there is none, defaults to dynamic.
- */
-const retrieveExportType = parameters => {
-  logger.debug(`Retrieving export type from parameters: ${JSON.stringify(parameters)}`);
-  const exportType = parameters.find(param => param.name === 'exportType');
-
-  if (!exportType) {
-    return 'dynamic';
-  } else {
-    return exportType.valueCode;
+const checkInputUrlArray = inputParamArray => {
+  if (inputParamArray.length === 0) {
+    throw new BadRequestError('No inputUrl parameters were found.');
   }
+  inputParamArray.forEach(inputUrl => {
+    if (!inputUrl.part.find(p => p.name === 'url').valueUrl) {
```

Comment:
This is a bit insufficient to check for bad input such as: 
```
{
  "resourceType": "Parameters",
  "parameter": [
    {
      "name": "input",
      "part": [
        {
          "name": "url",
          "valueUrl": "http://localhost:3001/b4d9c2c8-2630-4b80-a9e3-16fe26156dae/Patient.ndjson"
        }
      ]
    },
    {
      "name": "input",
      "part": [
      ]
    }
  ]
}
```

This causes a 500 to be returned since it tried to fetch `valueUrl` on undefined. Perhaps do another check on if `find()` found the `url` part or not and report that back as bad requestion.

Reply:
I am pretty sure this is fixed- can you confirm?

Reply:
Hmmm... now it says that it creates the import request just fine (which isn't good) and when you check the bulkstatus url you get the `batch-response` bundle with an OperationOutcome that says "Internal System Error: '$import' request not processed."

I was expecting this to return with a 400 error on the initial `$import` request with an indication of malformed import manifest.

Reply:
After discussing on slack, this is fine for the error to come back in the bulkstatus endpoint. But for this situation it should return an error indicating the issue with parsing the import manifest.